src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -236,11 +236,30 @@ Namespace Microsoft.CodeAnalysis.Semantics
             For Each base In statement.LocalDeclarations
                 If base.Kind = BoundKind.LocalDeclaration Then
                     Dim declaration = DirectCast(base, BoundLocalDeclaration)
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, Create(declaration.InitializerOpt), _semanticModel, declaration.Syntax))
+                    Dim initializer As IVariableInitializer = Nothing
+                    If declaration.InitializerOpt IsNot Nothing Then
+                        Debug.Assert(TypeOf declaration.Syntax Is ModifiedIdentifierSyntax)
+                        Dim initializerValue As IOperation = Create(declaration.InitializerOpt)
+                        Dim variableDeclaratorSyntax = DirectCast(declaration.Syntax.Parent, VariableDeclaratorSyntax)
```

Comment:
>DirectCast(declaration.Syntax.Parent, VariableDeclaratorSyntax) [](start = 55, length = 63)

The code would be more robust if this assumption was verified.

Reply:
Sure, I can add a kind check around this.